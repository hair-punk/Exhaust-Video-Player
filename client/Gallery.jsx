import React from 'react';
import styled from 'styled-components'
import OverflowScrolling from 'react-overflow-scrolling'

const ThumbNails =styled.img`
  height:71px;
  width:85px;
`;
const Bar = styled.div`
  overflow:auto;
  white-space:nowrap;
  opacity:1;
  background-color:#182937
  &::-webkit-scrollbar{
    width:15px
    height:15px
  }
  &::-webkit-scrollbar-thumb:horizontal{
    background-color:rgba( 103, 193, 245, .2 );
    border-radius:6px;
    width:10px;
    height:10px
    &:hover{
      background-color:#66C0F4
    }
  }
`;

class Gallery extends React.Component{
  constructor(props){
    super(props);
    this.state={
    }
  }
  componentDidMount(){
  }
  render(){
  var thumburls = this.props.thumbnailUrls.map((val,index)=><ThumbNails src={val} onClick={this.props.videoClickFunction.bind(null, index)}/>);
  var photourls = this.props.photoUrls.map((val, index)=><ThumbNails src = {val} onClick={this.props.photoClickFunction.bind(null, index)}/>);
  return (<OverflowScrolling >
    <Bar className="gallery-bar overflow-scrolling" >{thumburls}{photourls}{thumburls}{photourls}</Bar>
    </OverflowScrolling>);
  }
}

export default Gallery;