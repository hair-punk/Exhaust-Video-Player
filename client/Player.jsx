import React from 'react';
import ReactPlayer from 'react-player'
import styled from 'styled-components'
import Gallery from './Gallery.jsx'

const MediaContainer = styled.div`
  width:600px;
  height:450px;
`;
const Photo = styled.img`
  width:600px;
  height:360px;
  display:block;
`;
class Player extends React.Component {
  constructor(props){
    super(props);
    this.state={
      currentlyPlaying:'',
      currentPhoto:'',
      showPhoto:false,
      playing:true
    };
    this.videoClicked = this.videoClicked.bind(this);
    this.photoClicked = this.photoClicked.bind(this);
    this.componentDidMount = this.componentDidMount.bind(this)
  }
  videoClicked(index){
    this.setState((state,props)=>(
      {
        currentlyPlaying:this.props.videoUrls[index],
        showVideoThumbnail:true,
        showPhoto:false,
        playing:true
    }))
  }
  photoClicked(index){
    this.setState((state,props)=>(
      {
        showPhoto:true,
        currentPhoto:this.props.photoUrls[index],
        playing:false
      }
    ))
  }
  componentDidMount(){
    this.setState((state,props)=>(
      {
        currentlyPlaying:this.props.videoUrls[0],
        showPhoto:false,
        playing:true
      }
    ))
  }
  render(){
    if(this.state.showPhoto){
      return(<MediaContainer><Photo src={this.state.currentPhoto} />
        <Gallery thumbnailUrls={this.props.thumbnailUrls} photoUrls={this.props.photoUrls} videoClickFunction={this.videoClicked} photoClickFunction = {this.photoClicked}/>
        </MediaContainer>)
    }else
    return (<MediaContainer>
      <ReactPlayer style={{backgroundColor:'#182937'}}width="600px" url={this.state.currentlyPlaying} controls={true} volume={0.15} playing={this.state.playing} muted={true} file={{forceVideo:true}}/>
        <Gallery thumbnailUrls={this.props.thumbnailUrls} photoUrls={this.props.photoUrls} videoClickFunction={this.videoClicked} photoClickFunction = {this.photoClicked}/>
        </MediaContainer>)
  }
}
export default Player