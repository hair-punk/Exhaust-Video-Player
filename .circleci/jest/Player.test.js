import React from 'react';
import Player from '../../client/Player.jsx';
import Gallery from '../../client/Gallery.jsx'
import renderer from 'react-test-renderer';
import { shallow, mount, render} from 'enzyme';
import expect from 'expect';
// import 'jest-styled-components'
import {find, findAll} from 'styled-components/test-utils';
import toJson from 'enzyme-to-json'
import ReactPlayer from 'react-player';
import MediaContainer from '../../client/Player'

var videofilelinks = [];
for(var x = 0;x < 3;x++){
videofilelinks.push('video link' +x)
}
var photofilelinks = [];
for(var x =0;x < 3;x++){
  photofilelinks.push('photo link' +x)
}
var photothumblinks = [];
for(var x =0;x < 3;x++){
  photothumblinks.push('photo link' +x)
}

//isolation tests
describe('Player isolation testing', ()=>{
  const tree = mount((<Player videoUrls={videofilelinks} thumbnailUrls={photothumblinks} photoUrls={photofilelinks}/>))

    it('Player should match isolation snapshot',()=>{
      expect(toJson(tree)).toMatchSnapshot();
    })
    it('player link props should match passed in values',()=>{
      for(var x = 0;x < 3;x++){
        expect(tree.props().videoUrls[x]).toEqual(videofilelinks[x]);
      }
      for(var x = 0;x < 3;x++){
        expect(tree.props().photoUrls[x]).toEqual(photofilelinks[x]);
      }
      for(var x = 0;x < 3;x++){
        expect(tree.props().thumbnailUrls[x]).toEqual(photothumblinks[x]);
      }
    })
    it('should only have one ReactPlayer component upon initial render', ()=>{
      expect(tree.find(ReactPlayer).exists()).toEqual(true);
      expect(tree.find(ReactPlayer).length).toEqual(1);
    })
    it('should only have one gallery component upon initial render',()=>{
      expect(tree.find(Gallery).exists()).toEqual(true)
      expect(tree.find(Gallery).length).toEqual(1);
    })
    it('should have only one MediaContainer component upon initial render', ()=>{
      expect(tree.find(MediaContainer).exists()).toEqual(true);
      expect(tree.find(MediaContainer).length).toEqual(1);
    })
    it('should have a currentlyPlaying state that upon initial render, is set to \'\' ',()=>{
      expect(tree.state().currentlyPlaying).toEqual('')
    })
    it('should have a currentPhoto state that defaults to \'\' ',()=>{
      expect(tree.state().currentPhoto).toEqual('');
    })
    it('should have a showPhoto prop state defaults to false',()=>{
      expect(tree.state().showPhoto).toEqual(false);
    })
    it('should have a playing prop state defaults to true',()=>{
      expect(tree.state().playing).toEqual(true);
    })
})
