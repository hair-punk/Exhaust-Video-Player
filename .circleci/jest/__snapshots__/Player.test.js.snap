// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player isolation testing Player should match isolation snapshot 1`] = `
<Player
  photoUrls={
    Array [
      "photo link0",
      "photo link1",
      "photo link2",
    ]
  }
  thumbnailUrls={
    Array [
      "photo link0",
      "photo link1",
      "photo link2",
    ]
  }
  videoUrls={
    Array [
      "video link0",
      "video link1",
      "video link2",
    ]
  }
>
  <Player__MediaContainer>
    <div
      className="Player__MediaContainer-vjr722-0 jzcupq"
    >
      <ReactPlayer
        config={
          Object {
            "dailymotion": Object {
              "params": Object {
                "api": 1,
                "endscreen-enable": false,
              },
              "preload": false,
            },
            "facebook": Object {
              "appId": "1309697205772819",
            },
            "file": Object {
              "attributes": Object {},
              "dashVersion": "2.9.2",
              "forceAudio": false,
              "forceDASH": false,
              "forceHLS": false,
              "forceVideo": false,
              "hlsOptions": Object {},
              "hlsVersion": "0.10.1",
              "tracks": Array [],
            },
            "mixcloud": Object {
              "options": Object {
                "hide_cover": 1,
              },
            },
            "soundcloud": Object {
              "options": Object {
                "buying": false,
                "download": false,
                "liking": false,
                "sharing": false,
                "show_comments": false,
                "show_playcount": false,
                "visual": true,
              },
            },
            "twitch": Object {
              "options": Object {},
            },
            "vimeo": Object {
              "playerOptions": Object {
                "autopause": false,
                "byline": false,
                "portrait": false,
                "title": false,
              },
              "preload": false,
            },
            "wistia": Object {
              "options": Object {},
            },
            "youtube": Object {
              "embedOptions": Object {},
              "playerVars": Object {
                "iv_load_policy": 3,
                "modestbranding": 1,
                "playsinline": 1,
                "rel": 0,
                "showinfo": 0,
              },
              "preload": false,
            },
          }
        }
        controls={true}
        file={
          Object {
            "forceVideo": true,
          }
        }
        height="360px"
        light={false}
        loop={false}
        muted={true}
        onBuffer={[Function]}
        onBufferEnd={[Function]}
        onDisablePIP={[Function]}
        onDuration={[Function]}
        onEnablePIP={[Function]}
        onEnded={[Function]}
        onError={[Function]}
        onPause={[Function]}
        onPlay={[Function]}
        onProgress={[Function]}
        onReady={[Function]}
        onSeek={[Function]}
        onStart={[Function]}
        pip={false}
        playbackRate={1}
        playing={true}
        playsinline={false}
        progressInterval={1000}
        style={
          Object {
            "backgroundColor": "#182937",
          }
        }
        url=""
        volume={0.15}
        width="600px"
        wrapper="div"
      >
        <div
          file={
            Object {
              "forceVideo": true,
            }
          }
          style={
            Object {
              "backgroundColor": "#182937",
              "height": "360px",
              "width": "600px",
            }
          }
        />
      </ReactPlayer>
      <Gallery
        photoClickFunction={[Function]}
        photoUrls={
          Array [
            "photo link0",
            "photo link1",
            "photo link2",
          ]
        }
        thumbnailUrls={
          Array [
            "photo link0",
            "photo link1",
            "photo link2",
          ]
        }
        videoClickFunction={[Function]}
      >
        <OverflowScrolling>
          <ReactSwipe
            allowMouseEvents={false}
            innerRef={[Function]}
            onSwipeDown={[Function]}
            onSwipeEnd={[Function]}
            onSwipeLeft={[Function]}
            onSwipeMove={[Function]}
            onSwipeRight={[Function]}
            onSwipeStart={[Function]}
            onSwipeUp={[Function]}
            onWheel={[Function]}
            tagName="div"
            tolerance={0}
          >
            <div
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              onWheel={[Function]}
            >
              <Gallery__Bar
                className="gallery-bar overflow-scrolling"
              >
                <div
                  className="Gallery__Bar-sc-12uyjs6-1 irczOr gallery-bar overflow-scrolling"
                >
                  <Gallery__ThumbNails
                    key="0"
                    onClick={[Function]}
                    src="photo link0"
                  >
                    <img
                      className="Gallery__ThumbNails-sc-12uyjs6-0 kbeSNN"
                      onClick={[Function]}
                      src="photo link0"
                    />
                  </Gallery__ThumbNails>
                  <Gallery__ThumbNails
                    key="1"
                    onClick={[Function]}
                    src="photo link1"
                  >
                    <img
                      className="Gallery__ThumbNails-sc-12uyjs6-0 kbeSNN"
                      onClick={[Function]}
                      src="photo link1"
                    />
                  </Gallery__ThumbNails>
                  <Gallery__ThumbNails
                    key="2"
                    onClick={[Function]}
                    src="photo link2"
                  >
                    <img
                      className="Gallery__ThumbNails-sc-12uyjs6-0 kbeSNN"
                      onClick={[Function]}
                      src="photo link2"
                    />
                  </Gallery__ThumbNails>
                  <Gallery__ThumbNails
                    key="0"
                    onClick={[Function]}
                    src="photo link0"
                  >
                    <img
                      className="Gallery__ThumbNails-sc-12uyjs6-0 kbeSNN"
                      onClick={[Function]}
                      src="photo link0"
                    />
                  </Gallery__ThumbNails>
                  <Gallery__ThumbNails
                    key="1"
                    onClick={[Function]}
                    src="photo link1"
                  >
                    <img
                      className="Gallery__ThumbNails-sc-12uyjs6-0 kbeSNN"
                      onClick={[Function]}
                      src="photo link1"
                    />
                  </Gallery__ThumbNails>
                  <Gallery__ThumbNails
                    key="2"
                    onClick={[Function]}
                    src="photo link2"
                  >
                    <img
                      className="Gallery__ThumbNails-sc-12uyjs6-0 kbeSNN"
                      onClick={[Function]}
                      src="photo link2"
                    />
                  </Gallery__ThumbNails>
                  <Gallery__ThumbNails
                    key="0"
                    onClick={[Function]}
                    src="photo link0"
                  >
                    <img
                      className="Gallery__ThumbNails-sc-12uyjs6-0 kbeSNN"
                      onClick={[Function]}
                      src="photo link0"
                    />
                  </Gallery__ThumbNails>
                  <Gallery__ThumbNails
                    key="1"
                    onClick={[Function]}
                    src="photo link1"
                  >
                    <img
                      className="Gallery__ThumbNails-sc-12uyjs6-0 kbeSNN"
                      onClick={[Function]}
                      src="photo link1"
                    />
                  </Gallery__ThumbNails>
                  <Gallery__ThumbNails
                    key="2"
                    onClick={[Function]}
                    src="photo link2"
                  >
                    <img
                      className="Gallery__ThumbNails-sc-12uyjs6-0 kbeSNN"
                      onClick={[Function]}
                      src="photo link2"
                    />
                  </Gallery__ThumbNails>
                  <Gallery__ThumbNails
                    key="0"
                    onClick={[Function]}
                    src="photo link0"
                  >
                    <img
                      className="Gallery__ThumbNails-sc-12uyjs6-0 kbeSNN"
                      onClick={[Function]}
                      src="photo link0"
                    />
                  </Gallery__ThumbNails>
                  <Gallery__ThumbNails
                    key="1"
                    onClick={[Function]}
                    src="photo link1"
                  >
                    <img
                      className="Gallery__ThumbNails-sc-12uyjs6-0 kbeSNN"
                      onClick={[Function]}
                      src="photo link1"
                    />
                  </Gallery__ThumbNails>
                  <Gallery__ThumbNails
                    key="2"
                    onClick={[Function]}
                    src="photo link2"
                  >
                    <img
                      className="Gallery__ThumbNails-sc-12uyjs6-0 kbeSNN"
                      onClick={[Function]}
                      src="photo link2"
                    />
                  </Gallery__ThumbNails>
                </div>
              </Gallery__Bar>
            </div>
          </ReactSwipe>
        </OverflowScrolling>
      </Gallery>
    </div>
  </Player__MediaContainer>
</Player>
`;
